From d317d03f5741fec599ea5d99c0b28bea8f2b53d7 Mon Sep 17 00:00:00 2001
From: Pavan Balaji <balaji@mcs.anl.gov>
Date: Sat, 27 Apr 2013 12:16:40 -0500
Subject: [PATCH 2/2] Cleanup confdb macros.

1. AC_RUN_LOG is no longer an autoconf internal macro (it might have been
at some point).  We can use it directly.

2. Replace PAC_RUNLOG_IFELSE to use autoconf-native macros, plus some
minor additional work.  Get rid of PAC_COMMAND_IFELSE and
PAC_EVAL_IFELSE, which seem to be identical to PAC_RUNLOG_IFELSE.  One
option is to just point them to PAC_RUNLOG_IFELSE, but this commit
does not do that, since they seem to be unused.  Other projects that
use these macros can just point to PAC_RUNLOG_IFELSE if needed.

3. We no longer need PAS_VAR_COPY since AS_VAR_COPY is present in
newer autoconf versions.

4. PAC_VAR_PUSHFLAG and PAC_VAR_POPFLAG are duplicates of
PAC_PUSH_FLAG and PAC_POP_FLAG.  Also, unlike the PAC_VAR_ macro
variants, the flag is not updated by the PAC_PUSH/PAC_POP macros, so
the older value is retained.

5. The AM_IGNORE macro provided in confdb only works for a patched
automake, with no patch available to use.  We should either have both
or neither.

6. aclocal_atomic.m4 is now replaced by OPA.

7. The old F77 macros were kept around as backup when we moved to the
new macros (though this is really the job of the VCS).  It's been
several years since that was done.  We can safely remove the f77old.m4
file.  It is still in git if we ever need to go back.

8. Other general cleanup.

No reviewer.
---
 confdb/aclocal.m4            |   5 -
 confdb/aclocal_am.m4         |   6 -
 confdb/aclocal_atomic.m4     | 227 -------------------------
 confdb/aclocal_attr_alias.m4 |   8 +-
 confdb/aclocal_cc.m4         |   2 +-
 confdb/aclocal_cxx.m4        |   2 +-
 confdb/aclocal_f77.m4        |  38 ++---
 confdb/aclocal_f77old.m4     | 389 -------------------------------------------
 confdb/aclocal_fc.m4         |  10 +-
 confdb/aclocal_make.m4       |   2 -
 confdb/aclocal_runlog.m4     | 246 +++------------------------
 11 files changed, 50 insertions(+), 885 deletions(-)
 delete mode 100644 confdb/aclocal_am.m4
 delete mode 100644 confdb/aclocal_atomic.m4
 delete mode 100644 confdb/aclocal_f77old.m4

diff --git a/confdb/aclocal.m4 b/confdb/aclocal.m4
index 488feaa4..36fb15c9 100644
--- a/confdb/aclocal.m4
+++ b/confdb/aclocal.m4
@@ -1,18 +1,13 @@
 dnl This version of aclocal.m4 simply includes all of the individual
 dnl components
-builtin(include,aclocal_am.m4)
 builtin(include,aclocal_bugfix.m4)
 builtin(include,aclocal_cache.m4)
 builtin(include,aclocal_cc.m4)
-builtin(include,aclocal_atomic.m4)
 dnl aclocal_cross.m4 uses autoconf features dated back to 2.13.
 dnl too old to be useful, 07/14/2010.
 dnl builtin(include,aclocal_cross.m4)
 builtin(include,aclocal_cxx.m4)
 builtin(include,aclocal_f77.m4)
-dnl aclocal_f77old.m4 contains PAC_PROG_F77_CMDARGS which is NOT used in MPICH
-dnl but it is still used by other packages, so leave it in confdb.
-dnl builtin(include,aclocal_f77old.m4)
 builtin(include,aclocal_util.m4)
 builtin(include,aclocal_subcfg.m4)
 builtin(include,aclocal_make.m4)
diff --git a/confdb/aclocal_am.m4 b/confdb/aclocal_am.m4
deleted file mode 100644
index 4d3b0de2..00000000
--- a/confdb/aclocal_am.m4
+++ /dev/null
@@ -1,6 +0,0 @@
-dnl AM_IGNORE is an extension that tells (a patched) automake not to
-dnl include the specified AC_SUBST variable in the Makefile.in that
-dnl automake generates.  We don't use AC_DEFUN, since aclocal will 
-dnl then complain that AM_IGNORE is a duplicate (if you are using the
-dnl patched automake/aclocal).
-m4_ifdef([AM_IGNORE],[],[m4_define([AM_IGNORE],[])])
diff --git a/confdb/aclocal_atomic.m4 b/confdb/aclocal_atomic.m4
deleted file mode 100644
index 4e96cfbf..00000000
--- a/confdb/aclocal_atomic.m4
+++ /dev/null
@@ -1,227 +0,0 @@
-dnl /*D PAC_C_MEMATOMIC - Try and determine how to implement memory-atomic
-dnl   operations with the selected C compiler
-dnl
-dnl Synopsis:
-dnl PAC_C_MEMATOMIC
-dnl
-dnl Notes:
-dnl Defines names of the following form
-dnl + HAVE_GCC_ASM_AND_X86_{MFENCE,LFENCE,SFENCE} - gcc __asm__ will issue
-dnl    mfence, lfence, or sfence
-dnl . HAVE___ASM_AND_X86_{MFENCE,LFENCE,SFENCE} - __asm _emit will issue
-dnl    mfence, lfence, or sfence
-dnl . HAVE_ASM_AND_X86_{MFENCE,LFENCE,SFENCE} - asm("...") will issue
-dnl    mfence, lfence, or sfence
-dnl . HAVE__INTERLOCKEDEXCHANGE - _InterlockedExchange intrinsic is available
-dnl    (IA64)
-dnl . HAVE_GCC_ASM_SPARC_MEMBAR - gcc __asm__ will issue SPARC architecture
-dnl    memory barrier instruction
-dnl . HAVE_SOLARIS_ASM_SPARC_MEMBAR - Solaris asm() will issue SPARC 
-dnl    architecture memory barrier instruction
-dnl . HAVE_GCC_ASM_SPARC_STBAR - gcc __asm__ will issue stbar
-dnl - HAVE_SOLARIS_ASM_SPARC_STBAR - Solaris __asm() will issue stbar
-dnl 
-dnl D*/
-AC_DEFUN([PAC_C_MEMATOMIC],[
-AC_CACHE_CHECK([for x86 mfence instruction using __asm__],
-    pac_cv_have_gcc_asm_and_x86_mfence,[
-AC_TRY_RUN([
-int main(int argc, char **argv)
-{
-    __asm__ __volatile__  ( ".byte 0x0f, 0xae, 0xf0" ::: "memory" );
-    exit(0);
-}
-],
-pac_cv_have_gcc_asm_and_x86_mfence=yes,pac_cv_have_gcc_asm_and_x86_mfence=no)])
-
-if test "$pac_cv_have_gcc_asm_and_x86_mfence" = "yes" ; then
-    AC_DEFINE(HAVE_GCC_ASM_AND_X86_MFENCE, 1, [Define if using gcc on a x86 system with the mfence instruction])
-fi
-
-AC_CACHE_CHECK([for x86 sfence instruction using __asm__],
-    pac_cv_have_gcc_asm_and_x86_sfence,[
-AC_TRY_RUN([
-int main(int argc, char **argv)
-{
-    __asm__ __volatile__  ( ".byte 0x0f, 0xae, 0xf8" ::: "memory" );
-    exit(0);
-}
-],
-pac_cv_have_gcc_asm_and_x86_sfence=yes,pac_cv_have_gcc_asm_and_x86_sfence=no)])
-
-if test "$pac_cv_have_gcc_asm_and_x86_sfence" = "yes" ; then
-    AC_DEFINE(HAVE_GCC_ASM_AND_X86_SFENCE, 1, [Define if using gcc on a x86 system with the sfence instruction])
-fi
-
-AC_CACHE_CHECK([for x86 lfence instruction using __asm__],
-    pac_cv_have_gcc_asm_and_x86_lfence,[
-AC_TRY_RUN([
-int main(int argc, char **argv)
-{
-    __asm__ __volatile__  ( ".byte 0x0f, 0xae, 0xe8" ::: "memory" );
-    exit(0);
-}
-],
-pac_cv_have_gcc_asm_and_x86_lfence=yes,pac_cv_have_gcc_asm_and_x86_lfence=no)])
-
-if test "$pac_cv_have_gcc_asm_and_x86_lfence" = "yes" ; then
-    AC_DEFINE(HAVE_GCC_ASM_AND_X86_LFENCE, 1, [Define if using gcc on a x86 system with the lfence instruction])
-fi
-
-dnl Some compilers, like icc, may want __asm _emit
-AC_CACHE_CHECK([for x86 mfence instruction using __asm],
-     pac_cv_have___asm_and_x86_mfence,[
-AC_TRY_RUN([
-int main(int argc, char **argv)
-{
-    __asm _emit 0x0f __asm _emit 0xae __asm _emit 0xf0 ;
-    exit(0);
-}
-],
-pac_cv_have___asm_and_x86_mfence=yes,pac_cv_have___asm_and_x86_mfence=no)])
-
-if test "$pac_cv_have___asm_and_x86_mfence" = "yes" ; then
-    AC_DEFINE(HAVE___ASM_AND_X86_MFENCE, 1, [Define if using __asm on a x86 system with the mfence instruction])
-fi
-
-AC_CACHE_CHECK([for x86 sfence instruction using __asm],
-    pac_cv_have___asm_and_x86_sfence,[
-AC_TRY_RUN([
-int main(int argc, char **argv)
-{
-    __asm sfence ;
-    exit(0);
-}
-],
-pac_cv_have___asm_and_x86_sfence=yes,pac_cv_have___asm_and_x86_sfence=no)])
-
-if test "$pac_cv_have___asm_and_x86_sfence" = "yes" ; then
-    AC_DEFINE(HAVE___ASM_AND_X86_SFENCE, 1, [Define if using __asm on a x86 system with the sfence instruction])
-fi
-
-AC_CACHE_CHECK([for x86 lfence instruction using __asm],
-    pac_cv_have___asm_and_x86_lfence,[
-AC_TRY_RUN([
-int main(int argc, char **argv)
-{
-    __asm _emit 0x0f __asm _emit 0xae __asm _emit 0xe8 ;
-    exit(0);
-}
-],
-pac_cv_have___asm_and_x86_lfence=yes,pac_cv_have___asm_and_x86_lfence=no)])
-
-if test "$lac_cv_have___asm_and_x86_lfence" = "yes" ; then
-    AC_DEFINE(HAVE___ASM_AND_X86_LFENCE, 1, [Define if using __asm on a x86 system with the lfence instruction])
-fi
-
-dnl 
-dnl Some compilers, such as pgcc, may require additional arguments.
-dnl pgcc may need -Masmkeyword flag.  We may want to try this with and 
-dnl without adding -Masmkeyword to CFLAGS
-
-AC_CACHE_CHECK([for x86 mfence instruction using asm()],
-    pac_cv_have_asm_and_x86_mfence,[
-AC_TRY_RUN([
-int main(int argc, char **argv)
-{
-    asm("_emit 0x0f __asm _emit 0xae __asm _emit 0xf0");
-    exit(0);
-}
-],
-pac_cv_have_asm_and_x86_mfence=yes,pac_cv_have_asm_and_x86_mfence=no)])
-
-if test "$pac_cv_have_asm_and_x86_mfence" = "yes" ; then
-    AC_DEFINE(HAVE_ASM_AND_X86_MFENCE, 1, [Define if using asm() on a x86 system with the mfence instruction])
-fi
-
-AC_CACHE_CHECK([for x86 sfence instruction using asm()],
-    pac_cv_have_asm_and_x86_sfence,[
-AC_TRY_RUN([
-int main(int argc, char **argv)
-{
-    asm("sfence");
-    exit(0);
-}
-],
-pac_cv_have_asm_and_x86_sfence=yes,pac_cv_have_asm_and_x86_sfence=no)])
-
-if test "$pac_cv_have_asm_and_x86_sfence" = "yes" ; then
-    AC_DEFINE(HAVE_ASM_AND_X86_SFENCE, 1, [Define if using asm() on a x86 system with the sfence instruction])
-fi
-
-AC_CACHE_CHECK([for x86 lfence instruction using asm()],
-    pac_cv_have_asm_and_x86_lfence,[
-AC_TRY_RUN([
-int main(int argc, char **argv)
-{
-    asm("_emit 0x0f __asm _emit 0xae __asm _emit 0xe8");
-    exit(0);
-}
-],
-pac_cv_have_asm_and_x86_lfence=yes,pac_cv_have_asm_and_x86_lfence=no)])
-
-if test "$pac_cv_have_asm_and_x86_lfence" = "yes" ; then
-    AC_DEFINE(HAVE_ASM_AND_X86_LFENCE, 1, [Define if using asm() on a x86 system with the lfence instruction])
-fi
-
-AC_CACHE_CHECK([for _InterlockedExchange intrinsic],
-    pac_cv_have__InterlockedExchange,[
-AC_TRY_RUN([
-int main(int argc, char **argv)
-{
-    unsigned long lock, *lock_ptr;
-    lock_ptr = &lock;
-    _InterlockedExchange(lock_ptr, 1);
-    exit(0);
-}
-],
-pac_cv_have__InterlockedExchange=yes,pac_cv_have__InterlockedExchange=no)])
-
-if test "$pac_cv_have__InterlockedExchange" = "yes" ; then
-    AC_DEFINE(HAVE__INTERLOCKEDEXCHANGE, 1, [Define if _InterlockedExchange intrinsic is available])
-fi
-
-AC_CACHE_CHECK([for SPARC membar instruction with gcc],
-    pac_cv_gcc_sparc_membar,[
-AC_TRY_RUN([
-int main(int argc, char **argv){
-    __asm__ __volatile__ ( "membar #StoreLoad | #StoreStore" : : : "memory" );
-    exit(0);
-}],pac_cv_gcc_sparc_membar=yes,pac_cv_gcc_sparc_membar=no)])
-if test "$pac_cv_gcc_sparc_membar" = yes ; then
-    AC_DEFINE(HAVE_GCC_ASM_SPARC_MEMBAR,1,[Define if gcc asm membar supported])
-fi
-
-AC_CACHE_CHECK([for SPARC membar instruction with Solaris C],
-    pac_cv_solaris_sparc_membar,[
-AC_TRY_RUN([
-int main(int argc, char **argv){
-    __asm ( "membar #StoreLoad | #StoreStore");
-    exit(0);
-}],pac_cv_solaris_sparc_membar=yes,pac_cv_solaris_sparc_membar=no)])
-if test "$pac_cv_solaris_sparc_membar" = yes ; then
-    AC_DEFINE(HAVE_SOLARIS_ASM_SPARC_MEMBAR,1,[Define if solaris asm membar supported])
-fi
-
-AC_CACHE_CHECK([for SPARC stbar instruction with gcc],
-    pac_cv_gcc_sparc_stbar,[
-AC_TRY_RUN([
-int main(int argc, char **argv){
-    __asm__ __volatile__ ( "stbar" : : : "memory" );
-    exit(0);
-}],pac_cv_gcc_sparc_stbar=yes,pac_cv_gcc_sparc_stbar=no)])
-if test "$pac_cv_gcc_sparc_stbar" = yes ; then
-    AC_DEFINE(HAVE_GCC_ASM_SPARC_STBAR,1,[Define if gcc asm stbar supported])
-fi
-
-AC_CACHE_CHECK([for SPARC stbar instruction with Solaris C],
-    pac_cv_solaris_sparc_stbar,[
-AC_TRY_RUN([
-int main(int argc, char **argv){
-    __asm ( "stbar" );
-    exit(0);
-}],pac_cv_solaris_sparc_stbar=yes,pac_cv_solaris_sparc_stbar=no)])
-if test "$pac_cv_solaris_sparc_stbar" = yes ; then
-    AC_DEFINE(HAVE_SOLARIS_ASM_SPARC_STBAR,1,[Define if solaris asm stbar supported])
-fi
-])
\ No newline at end of file
diff --git a/confdb/aclocal_attr_alias.m4 b/confdb/aclocal_attr_alias.m4
index 32f01654..33db3293 100644
--- a/confdb/aclocal_attr_alias.m4
+++ b/confdb/aclocal_attr_alias.m4
@@ -148,7 +148,7 @@ int same_addrs2( void *ptr )
     ])
 ],[
     rm -f pac_conftest_other.$OBJEXT
-    PAC_RUNLOG([cp conftest.$OBJEXT pac_conftest_other.$OBJEXT])
+    AC_RUN_LOG([cp conftest.$OBJEXT pac_conftest_other.$OBJEXT])
     test -s pac_conftest_other.$OBJEXT && pac_c_attr_alias_other=yes
 dnl     cp conftest.$ac_ext pac_conftest_other.$ac_ext
 dnl     echo
@@ -196,7 +196,7 @@ extern int same_addrs2( void *ptr );
         ])
     ],[
         rm -f pac_conftest_main$EXEEXT
-        PAC_RUNLOG([cp conftest$EXEEXT pac_conftest_main$EXEEXT])
+        AC_RUN_LOG([cp conftest$EXEEXT pac_conftest_main$EXEEXT])
         test -x pac_conftest_main$EXEEXT && pac_c_attr_alias_main=yes
 dnl         cp conftest.$ac_ext pac_conftest_main.$ac_ext
 dnl         echo
@@ -371,7 +371,7 @@ dnl
 AC_LANG_PUSH([C])
 AC_COMPILE_IFELSE([AC_LANG_SOURCE([])],[
     cp conftest.$ac_ext pac_conftest.c
-    PAC_RUNLOG([mv conftest.$OBJEXT pac_conftest.$OBJEXT])
+    AC_RUN_LOG([mv conftest.$OBJEXT pac_conftest.$OBJEXT])
     PAC_PUSH_FLAG([LIBS])
     LIBS="pac_conftest.$OBJEXT $LIBS"
     AC_LANG_PUSH([Fortran]) dnl AC_LANG_PUSH([Fortran 77])
@@ -429,7 +429,7 @@ extern mpif_cmblk_t   mpifcmb_ __attribute__ ((alias("mpifcmbr")));
             ])
         ],[
             cp conftest.$ac_ext pac_conftest.c
-            PAC_RUNLOG([mv conftest.$OBJEXT pac_conftest.$OBJEXT])
+            AC_RUN_LOG([mv conftest.$OBJEXT pac_conftest.$OBJEXT])
 	    PAC_PUSH_FLAG([LIBS])
             LIBS="pac_conftest.$OBJEXT $LIBS"
             AC_LANG_PUSH([Fortran]) dnl AC_LANG_PUSH([Fortran 77])
diff --git a/confdb/aclocal_cc.m4 b/confdb/aclocal_cc.m4
index 98fdc347..06a51453 100644
--- a/confdb/aclocal_cc.m4
+++ b/confdb/aclocal_cc.m4
@@ -105,7 +105,7 @@ if test "$pac_result" = "yes" ; then
             int foo(void){return 0;}
         ])
     ],[
-        PAC_RUNLOG([mv conftest.$OBJEXT pac_conftest.$OBJEXT])
+        AC_RUN_LOG([mv conftest.$OBJEXT pac_conftest.$OBJEXT])
         saved_LIBS="$LIBS"
         LIBS="pac_conftest.$OBJEXT $LIBS"
 
diff --git a/confdb/aclocal_cxx.m4 b/confdb/aclocal_cxx.m4
index 279b6688..418d8b79 100644
--- a/confdb/aclocal_cxx.m4
+++ b/confdb/aclocal_cxx.m4
@@ -153,7 +153,7 @@ if test "$pac_result" = "yes" ; then
             int foo(void){return 0;}
         ])
     ],[
-        PAC_RUNLOG([mv conftest.$OBJEXT pac_conftest.$OBJEXT])
+        AC_RUN_LOG([mv conftest.$OBJEXT pac_conftest.$OBJEXT])
         saved_LIBS="$LIBS"
         LIBS="pac_conftest.$OBJEXT $LIBS"
 
diff --git a/confdb/aclocal_f77.m4 b/confdb/aclocal_f77.m4
index 92d310b0..38225e0e 100644
--- a/confdb/aclocal_f77.m4
+++ b/confdb/aclocal_f77.m4
@@ -64,7 +64,7 @@ AC_COMPILE_IFELSE([
         end
     ])
 ],[
-    PAC_RUNLOG([mv conftest.$OBJEXT f77conftest.$OBJEXT])
+    AC_RUN_LOG([mv conftest.$OBJEXT f77conftest.$OBJEXT])
     saved_LIBS="$LIBS"
     dnl  FLIBS is set by AC_F77_LIBRARY_LDFLAGS
     LIBS="f77conftest.$OBJEXT $FLIBS $LIBS"
@@ -96,7 +96,7 @@ if test  "$pac_found" != "yes" ; then
             AC_COMPILE_IFELSE([
                 AC_LANG_SOURCE([void ${call} ${sym}(int a) {}])
             ],[
-                PAC_RUNLOG([mv conftest.$OBJEXT cconftest.$OBJEXT])
+                AC_RUN_LOG([mv conftest.$OBJEXT cconftest.$OBJEXT])
                 saved_LIBS="$LIBS"
                 LIBS="cconftest.$OBJEXT $LIBS"
                 AC_LANG_PUSH([Fortran 77])
@@ -242,7 +242,7 @@ AC_COMPILE_IFELSE([
     ])
 ],[
     # pac_f77compile_ok=yes
-    PAC_RUNLOG([mv conftest.$OBJEXT pac_f77conftest.$OBJEXT])
+    AC_RUN_LOG([mv conftest.$OBJEXT pac_f77conftest.$OBJEXT])
     # Save original LIBS, prepend previously generated object file to LIBS
     saved_LIBS="$LIBS"
     LIBS="pac_f77conftest.$OBJEXT $FLIBS $LIBS"
@@ -349,7 +349,7 @@ int cisize_(char *i1p, char *i2p) {
     dnl Endof ac_lang_source
 ],[
     # pac_compile_ok=yes
-    PAC_RUNLOG([mv conftest.$OBJEXT pac_conftest.$OBJEXT])
+    AC_RUN_LOG([mv conftest.$OBJEXT pac_conftest.$OBJEXT])
     # Save LIBS and prepend object file to LIBS
     saved_LIBS="$LIBS"
     LIBS="pac_conftest.$OBJEXT $LIBS"
@@ -481,7 +481,7 @@ if test "$pac_result" = "yes" ; then
             end
         ])
     ],[
-        PAC_RUNLOG([mv conftest.$OBJEXT pac_conftest.$OBJEXT])
+        AC_RUN_LOG([mv conftest.$OBJEXT pac_conftest.$OBJEXT])
         saved_LIBS="$LIBS"
         LIBS="pac_conftest.$OBJEXT $LIBS"
 
@@ -541,10 +541,10 @@ AC_COMPILE_IFELSE([
     ])
 ],[
     # pac_f77compile_ok=yes
-    PAC_RUNLOG([mv conftest.$OBJEXT pac_f77conftest.$OBJEXT])
-    PAC_RUNLOG([test -d conftestdir || mkdir conftestdir])
-    PAC_RUNLOG([${AR-ar} ${AR_FLAGS-cr} conftestdir/libf77conftest.a pac_f77conftest.$OBJEXT])
-    PAC_RUNLOG([${RANLIB-ranlib} conftestdir/libf77conftest.a])
+    AC_RUN_LOG([mv conftest.$OBJEXT pac_f77conftest.$OBJEXT])
+    AC_RUN_LOG([test -d conftestdir || mkdir conftestdir])
+    AC_RUN_LOG([${AR-ar} ${AR_FLAGS-cr} conftestdir/libf77conftest.a pac_f77conftest.$OBJEXT])
+    AC_RUN_LOG([${RANLIB-ranlib} conftestdir/libf77conftest.a])
     # Save original LIBS, prepend previously generated object file to LIBS
     saved_LIBS="$LIBS"
     LIBS="-lf77conftest $LIBS"
@@ -595,7 +595,7 @@ ifelse([$1],[],[checkdir=f77tmpdir],[checkdir=$1;checkdir_is_given=yes])
 AC_CACHE_CHECK([for include directory flag for Fortran],
 pac_cv_prog_f77_has_incdir,[
 test -d $checkdir || mkdir $checkdir
-dnl PAC_RUNLOG([echo '       call sub()' > $checkdir/conftestf.h])
+dnl AC_RUN_LOG([echo '       call sub()' > $checkdir/conftestf.h])
 echo '       call sub()' > $checkdir/conftestf.h
 AC_LANG_PUSH([Fortran 77])
 saved_FFLAGS="$FFLAGS"
@@ -673,7 +673,7 @@ AC_COMPILE_IFELSE([
     AC_LANG_SOURCE([$2])
 ],[
     # pac_f77compile_ok=yes
-    PAC_RUNLOG([mv conftest.$OBJEXT pac_f77conftest.$OBJEXT])
+    AC_RUN_LOG([mv conftest.$OBJEXT pac_f77conftest.$OBJEXT])
     # Save original LIBS, prepend previously generated object file to LIBS
     saved_LIBS="$LIBS"
     LIBS="pac_f77conftest.$OBJEXT $FLIBS $LIBS"
@@ -732,7 +732,7 @@ AC_COMPILE_IFELSE([
     ])
 ],[
     # pac_f77compile_ok=yes
-    PAC_RUNLOG([mv conftest.$OBJEXT pac_f77conftest.$OBJEXT])
+    AC_RUN_LOG([mv conftest.$OBJEXT pac_f77conftest.$OBJEXT])
     # Save original LIBS, prepend previously generated object file to LIBS
     saved_LIBS="$LIBS"
     LIBS="pac_f77conftest.$OBJEXT $FLIBS $saved_LIBS"
@@ -833,7 +833,7 @@ dnl Use F77 as a linker to compile a Fortran main and C subprogram.
 if test "$pac_linkwithC" != "yes" ; then
     AC_LANG_PUSH([C])
     AC_COMPILE_IFELSE([],[
-        PAC_RUNLOG([mv conftest.$OBJEXT pac_conftest.$OBJEXT])
+        AC_RUN_LOG([mv conftest.$OBJEXT pac_conftest.$OBJEXT])
         saved_LIBS="$LIBS"
         LIBS="pac_conftest.$OBJEXT $saved_LIBS"
         AC_LANG_PUSH([Fortran 77])
@@ -852,7 +852,7 @@ dnl Use C as a linker and FLIBS to compile a Fortran main and C subprogram.
 if test "$pac_linkwithf77" != "yes" ; then
     AC_LANG_PUSH([Fortran 77])
     AC_COMPILE_IFELSE([],[
-        PAC_RUNLOG([mv conftest.$OBJEXT pac_f77conftest.$OBJEXT])
+        AC_RUN_LOG([mv conftest.$OBJEXT pac_f77conftest.$OBJEXT])
         saved_LIBS="$LIBS"
         LIBS="pac_f77conftest.$OBJEXT $FLIBS $saved_LIBS"
         AC_LANG_PUSH([C])
@@ -961,7 +961,7 @@ int $confname(int a) {
 }
     ])
 ],[
-    PAC_RUNLOG([mv conftest.$OBJEXT pac_conftest.$OBJEXT])
+    AC_RUN_LOG([mv conftest.$OBJEXT pac_conftest.$OBJEXT])
     saved_LIBS="$LIBS"
     AC_LANG_PUSH([Fortran 77])
     AC_LANG_CONFTEST([
@@ -1089,7 +1089,7 @@ dnl Use F77 as a linker to compile a Fortran main and C subprogram.
 if test "$pac_linkwithC" != "yes" ; then
     AC_LANG_PUSH([C])
     AC_COMPILE_IFELSE([],[
-        PAC_RUNLOG([mv conftest.$OBJEXT pac_conftest.$OBJEXT])
+        AC_RUN_LOG([mv conftest.$OBJEXT pac_conftest.$OBJEXT])
         saved_LIBS="$LIBS"
         LIBS="pac_conftest.$OBJEXT $saved_LIBS"
         AC_LANG_PUSH([Fortran 77])
@@ -1110,7 +1110,7 @@ dnl Use C as a linker and FLIBS to compile a Fortran main and C subprogram.
 if test "$pac_linkwithf77" != "yes" ; then
     AC_LANG_PUSH([Fortran 77])
     AC_COMPILE_IFELSE([],[
-        PAC_RUNLOG([mv conftest.$OBJEXT pac_f77conftest.$OBJEXT])
+        AC_RUN_LOG([mv conftest.$OBJEXT pac_f77conftest.$OBJEXT])
         saved_LIBS="$LIBS"
         LIBS="pac_f77conftest.$OBJEXT $FLIBS $saved_LIBS"
         AC_LANG_PUSH([C])
@@ -1277,7 +1277,7 @@ AC_COMPILE_IFELSE([
         end
     ])
 ],[
-    PAC_RUNLOG([mv conftest.$OBJEXT pac_f77conftest.$OBJEXT])
+    AC_RUN_LOG([mv conftest.$OBJEXT pac_f77conftest.$OBJEXT])
     saved_LIBS="$LIBS"
     # This test checks if Fortran init can be done in pure C environment,
     # i.e. no FLIBS in linking, so don't put FLIBS in LIBS below
@@ -1368,7 +1368,7 @@ void ftest_( $pac_mpi_fint *itrue, $pac_mpi_fint *ifalse )
 }
     ])
 ],[
-    PAC_RUNLOG([mv conftest.$OBJEXT pac_conftest.$OBJEXT])
+    AC_RUN_LOG([mv conftest.$OBJEXT pac_conftest.$OBJEXT])
     saved_LIBS="$LIBS"
     LIBS="pac_conftest.$OBJEXT $saved_LIBS"
     AC_LANG_PUSH([Fortran 77])
diff --git a/confdb/aclocal_f77old.m4 b/confdb/aclocal_f77old.m4
deleted file mode 100644
index 5d6e2d9e..00000000
--- a/confdb/aclocal_f77old.m4
+++ /dev/null
@@ -1,389 +0,0 @@
-dnl/*D
-dnl PAC_PROG_F77_CMDARGS - Determine how to access the command line from
-dnl Fortran 77
-dnl
-dnl Output Effects:
-dnl  The following variables are set:
-dnl.vb
-dnl    F77_GETARG         - Statement to get an argument i into string s
-dnl    F77_IARGC          - Routine to return the number of arguments
-dnl    FXX_MODULE         - Module command when using Fortran 90 compiler
-dnl    F77_GETARGDECL     - Declaration of routine used for F77_GETARG
-dnl    F77_GETARG_FFLAGS  - Flags needed when compiling/linking
-dnl    F77_GETARG_LDFLAGS - Flags needed when linking
-dnl.ve
-dnl If 'F77_GETARG' has a value, then that value and the values for these
-dnl other symbols will be used instead.  If no approach is found, all of these
-dnl variables will have empty values.
-dnl If no other approach works and a file 'f77argdef' is in the directory, 
-dnl that file will be sourced for the values of the above four variables.
-dnl
-dnl In most cases, you should add F77_GETARG_FFLAGS to the FFLAGS variable
-dnl and F77_GETARG_LDFLAGS to the LDFLAGS variable, to ensure that tests are
-dnl performed on the compiler version that will be used.
-dnl
-dnl 'AC_SUBST' is called for all six variables.
-dnl
-dnl One complication is that on systems with multiple Fortran compilers, 
-dnl some libraries used by one Fortran compiler may have been (mis)placed
-dnl in a common location.  We have had trouble with libg2c in particular.
-dnl To work around this, we test whether iargc etc. work first.  This
-dnl will catch most systems and will speed up the tests.
-dnl
-dnl Next, the libraries are only added if they are needed to complete a 
-dnl link; they aren''t added just because they exist.
-dnl
-dnl f77argdef
-dnl D*/
-dnl
-dnl Random notes
-dnl You can export the command line arguments from C to the g77 compiler
-dnl using
-dnl    extern char **__libc_argv;
-dnl    extern int  __libc_argc;
-dnl    f_setarg( __libc_argc, __libc_argv );
-dnl
-AC_DEFUN([PAC_PROG_F77_CMDARGS],[
-found_cached="yes"
-AC_MSG_CHECKING([for routines to access the command line from Fortran 77])
-AC_CACHE_VAL(pac_cv_prog_f77_cmdarg,
-[
-    AC_MSG_RESULT([searching...])
-    found_cached="no"
-    # First, we perform a quick check.  Does iargc and getarg work?
-    fxx_module="${FXX_MODULE:-}"
-    f77_getargdecl="${F77_GETARGDECL:-external getarg}"
-    f77_getarg="${F77_GETARG:-call GETARG(i,s)}"
-    f77_iargc="${F77_IARGC:-IARGC()}"
-    #    
-    # Grumble.  The Absoft Fortran compiler computes i - i as 0 and then
-    # 1.0 / 0 at compile time, even though the code may never be executed.
-    # What we need is a way to generate an error, so the second usage of i
-    # was replaced with f77_iargc.  
-    cat > conftest.f <<EOF
-        program main
-$fxx_module
-        integer i, j
-        character*20 s
-        $f77_getargdecl
-        i = 0
-        $f77_getarg
-        i=$f77_iargc
-        if (i .gt. 1) then
-            j = i - $f77_iargc
-            j = 1.0 / j
-        endif
-        end
-EOF
-    found_answer="no"
-    if test -z "$ac_fcompilelink" ; then
-        ac_fcompilelink="${F77-f77} -o conftest $FFLAGS $flags conftest.f $LDFLAGS $LIBS 1>&AC_FD_CC"
-    fi
-    AC_MSG_CHECKING([whether ${F77-f77} $flags $libs works with GETARG and IARGC])
-    if AC_TRY_EVAL(ac_fcompilelink) && test -x conftest ; then
-	# Check that cross != yes so that this works with autoconf 2.52
-	# Check that cross_compiling != yes so that this works with 
-	# autoconf 2.6x for some (but almost certainly not all)x
-	# Question: why do we test that this runs?  It looks like we
-	# needed this for some old Fortran compilers that produced
-	# executable code that then did not run.
-	if test "$ac_cv_prog_f77_cross" != "yes" -a \
-	        "$cross_compiling" != "yes" ; then
-	    if ./conftest >/dev/null 2>&1 ; then
-		found_answer="yes"
-	        FXX_MODULE="$fxx_module"
-		F77_GETARGDECL="$f77_getargdecl"
-		F77_GETARG="$f77_getarg"
-		F77_IARGC="$f77_iargc"
-		AC_MSG_RESULT(yes)
-     	    fi
-        fi
-    fi    
-    if test $found_answer = "no" ; then
-	AC_MSG_RESULT(no)
-    # Grumph.  Here are a bunch of different approaches
-    # We have several axes the check:
-    # Library to link with (none, -lU77 (HPUX), -lg2c (LINUX f77))
-    # PEPCF90 (Intel ifc)
-    # The first line is a dummy
-    # (we experimented with using a <space>, but this caused other 
-    # problems because we need <space> in the IFS)
-    trial_LIBS="0 -lU77 -lPEPCF90"
-    if test "$NOG2C" != "1" ; then
-        trial_LIBS="$trial_LIBS -lg2c"
-    fi
-    # Discard libs that are not availble:
-    save_IFS="$IFS"
-    # Make sure that IFS includes a space, or the tests that run programs
-    # may fail
-    IFS=" ""
-"
-    save_trial_LIBS="$trial_LIBS"
-    trial_LIBS=""
-    cat > conftest.f <<EOF
-        program main
-        end
-EOF
-    ac_fcompilelink_test='${F77-f77} -o conftest $FFLAGS conftest.f $LDFLAGS $libs $LIBS 1>&AC_FD_CC'
-    for libs in $save_trial_LIBS ; do
-	if test "$libs" = "0" ; then
-	    lib_ok="yes"
-        else
-	    AC_MSG_CHECKING([whether Fortran 77 links with $libs])
-	    if AC_TRY_EVAL(ac_fcompilelink_test) && test -x conftest ; then
-		AC_MSG_RESULT([yes])
-	        lib_ok="yes"
-	    else
-		AC_MSG_RESULT([no])
-	        lib_ok="no"
-	    fi
-	fi
-	if test "$lib_ok" = "yes" ; then
-	    trial_LIBS="$trial_LIBS
-$libs"
-        fi
-    done
-
-    # Options to use when compiling and linking
-    # +U77 is needed by HP Fortran to access getarg etc.
-    # The -N109 was used for getarg before we realized that GETARG
-    # was necessary with the (non standard conforming) Absoft compiler
-    # (Fortran is monocase; Absoft uses mixedcase by default)
-    # The -f is used by Absoft and is the compiler switch that folds 
-    # symbolic names to lower case.  Without this option, the compiler
-    # considers upper- and lower-case letters to be unique.
-    # The -YEXT_NAMES=LCS will cause external names to be output as lower
-    # case letter for Absoft F90 compilers (default is upper case)
-    # The first line is "<space><newline>, the space is important
-    # To make the Absoft f77 and f90 work together, we need to prefer the
-    # upper case versions of the arguments.  They also require libU77.
-    # -YCFRL=1 causes Absoft f90 to work with g77 and similar (f2c-based) 
-    # Fortran compilers
-    #
-    # Problem:  The Intel efc compiler hangs when presented with -N109 .
-    # The only real fix for this is to detect this compiler and exclude
-    # the test.  We may want to reorganize these tests so that if we
-    # can compile code without special options, we never look for them.
-    # 
-    using_intel_efc="no"
-    pac_test_msg=`$F77 -V 2>&1 | grep 'Intel(R) Fortran Itanium'`
-    if test "$pac_test_msg" != "" ; then
-	using_intel_efc="yes"
-    fi
-    if test "$using_intel_efc" = "yes" ; then
-        trial_FLAGS="000"
-    else
-        trial_FLAGS="000
--N109
--f
--YEXT_NAMES=UCS
--YEXT_NAMES=LCS
--YCFRL=1
-+U77"
-    fi
-    # Discard options that are not available:
-    # (IFS already saved above)
-    IFS=" ""
-"
-    save_trial_FLAGS="$trial_FLAGS"
-    trial_FLAGS=""
-    for flag in $save_trial_FLAGS ; do
-	if test "$flag" = " " -o "$flag" = "000" ; then
-	    opt_ok="yes"
-        else
-            PAC_F77_CHECK_COMPILER_OPTION($flag,opt_ok=yes,opt_ok=no)
-        fi
-	if test "$opt_ok" = "yes" ; then
-	    if test "$flag" = " " -o "$flag" = "000" ; then 
-		fflag="" 
-	    else 
-		fflag="$flag" 
-	    fi
-	    # discard options that don't allow mixed-case name matching
-	    cat > conftest.f <<EOF
-        program main
-        call aB()
-        end
-        subroutine Ab()
-        end
-EOF
-	    if test -n "$fflag" ; then flagval="with $fflag" ; else flagval="" ; fi
-	    AC_MSG_CHECKING([whether Fortran 77 routine names are case-insensitive $flagval])
-	    dnl we can use double quotes here because all is already
-            dnl evaluated
-            ac_fcompilelink_test="${F77-f77} -o conftest $fflag $FFLAGS conftest.f $LDFLAGS $LIBS 1>&AC_FD_CC"
-	    if AC_TRY_EVAL(ac_fcompilelink_test) && test -x conftest ; then
-	        AC_MSG_RESULT(yes)
-	    else
-	        AC_MSG_RESULT(no)
-	        opt_ok="no"
-            fi
-        fi
-        if test "$opt_ok" = "yes" ; then
-	    trial_FLAGS="$trial_FLAGS
-$flag"
-        fi
-    done
-    IFS="$save_IFS"
-    # Name of routines.  Since these are in groups, we use a case statement
-    # and loop until the end (accomplished by reaching the end of the
-    # case statement
-    # For one version of Nag F90, the names are 
-    # call f90_unix_MP_getarg(i,s) and f90_unix_MP_iargc().
-    trial=0
-    while test -z "$pac_cv_prog_f77_cmdarg" ; do
-        case $trial in 
-	0) # User-specified values, if any
-	   if test -z "$F77_GETARG" -o -z "$F77_IARGC" ; then 
-	       trial=`expr $trial + 1`
-	       continue 
-           fi
-           MSG="Using environment values of F77_GETARG etc."
-	   ;;
-	1) # Standard practice, uppercase (some compilers are case-sensitive)
-	   FXX_MODULE=""
-	   F77_GETARGDECL="external GETARG"
-	   F77_GETARG="call GETARG(i,s)"
-	   F77_IARGC="IARGC()"
-	   MSG="GETARG and IARGC"
-	   ;;
-	2) # Standard practice, lowercase
-	   FXX_MODULE=""
-           F77_GETARGDECL="external getarg"
-	   F77_GETARG="call getarg(i,s)"
-	   F77_IARGC="iargc()"
-	   MSG="getarg and iargc"
-	   ;;
-	3) # Posix alternative
-	   FXX_MODULE=""
-	   F77_GETARGDECL="external pxfgetarg"
-	   F77_GETARG="call pxfgetarg(i,s,l,ier)"
-	   F77_IARGC="ipxfargc()"
-	   MSG="pxfgetarg and ipxfargc"
-	   ;;
-	4) # Nag f90_unix_env module
-	   FXX_MODULE="        use f90_unix_env"
-	   F77_GETARGDECL=""
-	   F77_GETARG="call getarg(i,s)"
-	   F77_IARGC="iargc()"
-	   MSG="f90_unix_env module"
-	   ;;
-        5) # Nag f90_unix module
-	   FXX_MODULE="        use f90_unix"
-	   F77_GETARGDECL=""
-	   F77_GETARG="call getarg(i,s)"
-	   F77_IARGC="iargc()"
-	   MSG="f90_unix module"
-	   ;;
-	6) # user spec in a file
-	   if test -s f77argdef ; then
-		. ./f77argdef
-	       MSG="Using definitions in the file f77argdef"
-	   else
-	        trial=`expr $trial + 1`
-		continue
-	   fi
-	   ;;
-	7) # gfortran won't find getarg if it is marked as external 
-	   FXX_MODULE=""
-	   F77_GETARGDECL="intrinsic GETARG"
-	   F77_GETARG="call GETARG(i,s)"
-	   F77_IARGC="IARGC()"
-	   MSG="intrinsic GETARG and IARGC"
-	   ;;
-        *) # exit from while loop
-	   FXX_MODULE=""
-	   F77_GETARGDECL=""
-	   F77_GETARG=""
-	   F77_IARGC=""
-           break
-	   ;;
-	esac
-	# Create the program.  Make sure that we can run it.
-	# Force a divide-by-zero if there is a problem (but only at runtime!
-        # (the Absoft compiler does divide-by-zero at compile time)
-        cat > conftest.f <<EOF
-        program main
-$FXX_MODULE
-        integer i, j
-        character*20 s
-        $F77_GETARGDECL
-        i = 0
-        $F77_GETARG
-        i=$F77_IARGC
-        if (i .gt. 1) then
-            j = i - $F77_IARGC
-            j = 1.0 / j
-        endif
-        end
-EOF
-    #
-    # Now, try to find some way to compile and link that program, looping 
-    # over the possibilities of options and libraries
-        save_IFS="$IFS"
-        IFS=" ""
-"
-        for libs in $trial_LIBS ; do
-            if test -n "$pac_cv_prog_f77_cmdarg" ; then break ; fi
-	    if test "$libs" = " " -o "$libs" = "0" ; then libs="" ; fi
-            for flags in $trial_FLAGS ; do
-	        if test "$flags" = " " -o "$flags" = "000"; then flags="" ; fi
-                AC_MSG_CHECKING([whether ${F77-f77} $flags $libs works with $MSG])
-		IFS="$save_IFS"
-		dnl We need this here because we've fiddled with IFS
-	        ac_fcompilelink_test="${F77-f77} -o conftest $FFLAGS $flags conftest.f $LDFLAGS $libs $LIBS 1>&AC_FD_CC"
-		found_answer="no"
-                if AC_TRY_EVAL(ac_fcompilelink_test) && test -x conftest ; then
-		    if test "$ac_cv_prog_f77_cross" != "yes" -a \	 
-		            "$cross_compiling" != "yes" ; then
-			if ./conftest >/dev/null 2>&1 ; then
-			    found_answer="yes"
-			fi
-		    else 
-			found_answer="yes"
-		    fi
-                fi
-	        IFS=" ""
-"
-		if test "$found_answer" = "yes" ; then
-	            AC_MSG_RESULT([yes])
-		    pac_cv_prog_f77_cmdarg="$MSG"
-		    pac_cv_prog_f77_cmdarg_fflags="$flags"
-		    pac_cv_prog_f77_cmdarg_ldflags="$libs"
-		    break
-	        else
-                    AC_MSG_RESULT([no])
-		    echo "configure: failed program was:" >&AC_FD_CC
-                    cat conftest.f >&AC_FD_CC
-	        fi
-            done
-        done
-        IFS="$save_IFS"   
-	rm -f conftest.*
-        trial=`expr $trial + 1`   
-    done
-fi
-pac_cv_F77_GETARGDECL="$F77_GETARGDECL"
-pac_cv_F77_IARGC="$F77_IARGC"
-pac_cv_F77_GETARG="$F77_GETARG"
-pac_cv_FXX_MODULE="$FXX_MODULE"
-])
-if test "$found_cached" = "yes" ; then 
-    AC_MSG_RESULT([$pac_cv_prog_f77_cmdarg])
-elif test -z "$pac_cv_F77_IARGC" ; then
-    AC_MSG_WARN([Could not find a way to access the command line from Fortran 77])
-fi
-# Set the variable values based on pac_cv_prog_xxx
-F77_GETARGDECL="$pac_cv_F77_GETARGDECL"
-F77_IARGC="$pac_cv_F77_IARGC"
-F77_GETARG="$pac_cv_F77_GETARG"
-FXX_MODULE="$pac_cv_FXX_MODULE"
-F77_GETARG_FFLAGS="$pac_cv_prog_f77_cmdarg_fflags"
-F77_GETARG_LDFLAGS="$pac_cv_prog_f77_cmdarg_ldflags"
-AC_SUBST(F77_GETARGDECL)
-AC_SUBST(F77_IARGC)
-AC_SUBST(F77_GETARG)
-AC_SUBST(FXX_MODULE)
-AC_SUBST(F77_GETARG_FFLAGS)
-AC_SUBST(F77_GETARG_LDFLAGS)
-])
diff --git a/confdb/aclocal_fc.m4 b/confdb/aclocal_fc.m4
index 02a5d5da..716f085e 100644
--- a/confdb/aclocal_fc.m4
+++ b/confdb/aclocal_fc.m4
@@ -468,7 +468,7 @@ AC_COMPILE_IFELSE([
     ])
 ],[
     # pac_f77compile_ok=yes
-    PAC_RUNLOG([mv conftest.$OBJEXT pac_f77conftest.$OBJEXT])
+    AC_RUN_LOG([mv conftest.$OBJEXT pac_f77conftest.$OBJEXT])
     # Save original LIBS, prepend previously generated object file to LIBS
     saved_LIBS="$LIBS"
     LIBS="pac_f77conftest.$OBJEXT $LIBS"
@@ -589,7 +589,7 @@ int $confname( int a )
     ])
 ],[
     pac_compile_ok=yes
-    PAC_RUNLOG([mv conftest.$OBJEXT pac_conftest.$OBJEXT])
+    AC_RUN_LOG([mv conftest.$OBJEXT pac_conftest.$OBJEXT])
     # Save LIBS and prepend object file to LIBS
     saved_LIBS="$LIBS"
     LIBS="pac_conftest.$OBJEXT $LIBS"
@@ -678,7 +678,7 @@ if test "$pac_result" = "yes" ; then
             end
         ])
     ],[
-        PAC_RUNLOG([mv conftest.$OBJEXT pac_conftest.$OBJEXT])
+        AC_RUN_LOG([mv conftest.$OBJEXT pac_conftest.$OBJEXT])
         saved_LIBS="$LIBS"
         LIBS="pac_conftest.$OBJEXT $LIBS"
 
@@ -778,7 +778,7 @@ AC_MSG_CHECKING([for Fortran 90 compiler vendor])
 pac_cv_fc_vendor="unknown"
 for arg in --version -V -v ; do
     rm -f conftest.txt
-    PAC_RUNLOG([$FC $arg </dev/null >conftest.txt 2>&1])
+    AC_RUN_LOG([$FC $arg </dev/null >conftest.txt 2>&1])
     # Ignore the return code, because some compilers set the
     # return code to zero on invalid arguments and some to 
     # non-zero on success (with no files to compile)
@@ -1027,7 +1027,7 @@ int cisize_(char *i1p, char *i2p)
 }
     ])
 ],[
-    PAC_RUNLOG([mv conftest.$OBJEXT pac_conftest.$OBJEXT])
+    AC_RUN_LOG([mv conftest.$OBJEXT pac_conftest.$OBJEXT])
     pac_ccompile_ok=yes
 ],[
     pac_ccompile_ok=no
diff --git a/confdb/aclocal_make.m4 b/confdb/aclocal_make.m4
index 147e92e9..0f73e236 100644
--- a/confdb/aclocal_make.m4
+++ b/confdb/aclocal_make.m4
@@ -148,7 +148,6 @@ dnl
 dnl D*/
 AC_DEFUN([PAC_PROG_MAKE_VPATH],[
 AC_SUBST(VPATH)
-dnl AM_IGNORE(VPATH)
 AC_CACHE_CHECK([for virtual path format],
 pac_cv_prog_make_vpath,[
 AC_REQUIRE([PAC_PROG_MAKE_PROGRAM])
@@ -305,7 +304,6 @@ PAC_PROG_MAKE_INCLUDE
 PAC_PROG_MAKE_ALLOWS_COMMENTS
 PAC_PROG_MAKE_VPATH
 AC_SUBST(SET_CFLAGS)
-dnl AM_IGNORE(SET_CFLAGS)
 PAC_PROG_MAKE_SET_CFLAGS([SET_CFLAGS='CFLAGS='])
 if test "$pac_cv_prog_make_echos_dir" = "no" ; then
     AC_PROG_MAKE_SET
diff --git a/confdb/aclocal_runlog.m4 b/confdb/aclocal_runlog.m4
index 9b85db56..ebab6ba0 100644
--- a/confdb/aclocal_runlog.m4
+++ b/confdb/aclocal_runlog.m4
@@ -1,252 +1,46 @@
-dnl
-dnl PAC_RUN_LOG mimics _AC_RUN_LOG which is autoconf internal routine.
-dnl We also make sure PAC_RUN_LOG can be used in AS_IF, so the last
-dnl test command should have terminating ]), i.e. without newline before ]).
-dnl
-AC_DEFUN([PAC_RUNLOG],[
-{ AS_ECHO(["$as_me:$LINENO: $1"]) >&AS_MESSAGE_LOG_FD
-  (eval $1) 2>&AS_MESSAGE_LOG_FD
-  ac_status=$?
-  AS_ECHO(["$as_me:$LINENO: \$? = $ac_status"]) >&AS_MESSAGE_LOG_FD
-  test $ac_status = 0; }])
-dnl
-dnl PAC_COMMAND_IFELSE is written to replace AC_TRY_EVAL with added logging
-dnl to config.log, i.e. AC_TRY_EVAL does not log anything to config.log.
-dnl If autoconf provides AC_COMMAND_IFELSE or AC_EVAL_IFELSE,
-dnl AC_COMMAND_IFELSE dnl should be replaced by the official autoconf macros.
-dnl
-dnl PAC_COMMAND_IFELSE(COMMMAND,[ACTION-IF-RUN-OK],[ACTION-IF-RUN-FAIL])
-dnl
-AC_DEFUN([PAC_COMMAND_IFELSE],[
-dnl Should use _AC_DO_TOKENS but use AC_RUN_LOG instead
-dnl because _AC_XX is autoconf's undocumented macro.
-AS_IF([PAC_RUNLOG([$1])],[
-    $2
-],[
-    AS_ECHO(["$as_me: program exited with status $ac_status"]) >&AS_MESSAGE_LOG_FD
-    m4_ifvaln([$3],[
-        (exit $ac_status)
-        $3
-    ])
-])
-])
-dnl
-dnl
-dnl
-AC_DEFUN([PAC_EVAL_IFELSE],[
-dnl Should use _AC_DO_TOKENS but use AC_RUN_LOG instead
-dnl because _AC_XX is autoconf's undocumented macro.
-AS_IF([PAC_RUNLOG([$$1])],[
-    $2
-],[
-    AS_ECHO(["$as_me: program exited with status $ac_status"]) >&AS_MESSAGE_LOG_FD
-    m4_ifvaln([$3],[
-        (exit $ac_status)
-        $3
-    ])
-])
-])
-dnl
-dnl
-dnl
-AC_DEFUN([PAC_RUNLOG_IFELSE],[
-dnl pac_TESTLOG is the internal temporary logfile for this macro.
-pac_TESTLOG="pac_test.log"
-rm -f $pac_TESTLOG
-PAC_COMMAND_IFELSE([$1 > $pac_TESTLOG],[
-    ifelse([$2],[],[],[$2])
-],[
-    AS_ECHO(["*** $1 :"]) >&AS_MESSAGE_LOG_FD
-    cat $pac_TESTLOG >&AS_MESSAGE_LOG_FD
-    ifelse([$3],[],[],[$3])
-])
-rm -f $pac_TESTLOG
-])
-dnl
-dnl
-dnl
-dnl PAS_VAR_COPY -  A portable layer that mimics AS_VAR_COPY when it is not
-dnl                 defined as in older autoconf, e.g. 2.63 and older.
-dnl                 This macro is absolutely necessary, because AS_VAR_GET in
-dnl                 some newer autoconf, e.g. 2.64, seems to be totally broken,
-dnl                 or behave very different from older autoconf, i.e. 2.63.
-dnl
-AC_DEFUN([PAS_VAR_COPY],[
-m4_ifdef([AS_VAR_COPY], [AS_VAR_COPY([$1],[$2])], [$1=AS_VAR_GET([$2])])
-])
-dnl
-dnl
-dnl
-dnl PAC_VAR_PUSHVAL(VARNAME, [LastSavedValue]))
-dnl
-dnl Save the content of the shell variable, VARNAME, onto a stack.
-dnl The saved value of VARNAME is restorable with respect to the nesting
-dnl of the macro.
-dnl
-dnl The Last saved value of VARNAME on the stack is stored in shell variable
-dnl pac_LastSavedValueOf_$VARNAME if the 2nd argument is NOT supplied.
-dnl If the 2nd argument is present, the last saved value will be stored
-dnl in the 2nd argument instead.
-dnl
-dnl The First saved value of VARNAME on the stack is stored in shell variable
-dnl dnl pac_FirstSavedValueOf_$VARNAME.
-dnl
-AC_DEFUN([PAC_VAR_PUSHVAL],[
-# START of PUSHVAL
-dnl define local m4-name pac_stk_level.
-AS_VAR_PUSHDEF([pac_stk_level], [pac_stk_$1_level])
-AS_VAR_SET_IF([pac_stk_level],[
-    dnl autoconf < 2.64 does not have AS_VAR_ARITH, so use expr instead.
-    AS_VAR_SET([pac_stk_level], [`expr $pac_stk_level + 1`])
-],[
-    AS_VAR_SET([pac_stk_level], [0])
-])
-dnl AS_ECHO_N(["PUSHVAL: pac_stk_level = $pac_stk_level, "])
-dnl Save the content of VARNAME, i.e. $VARNAME, onto the stack.
-AS_VAR_SET([pac_stk_$1_$pac_stk_level],[$$1])
-AS_VAR_IF([pac_stk_level], [0], [
-    dnl Save the 1st pushed value of VARNAME as pac_FirstSavedValueOf_$VARNAME
-    PAS_VAR_COPY([pac_FirstSavedValueOf_$1],[pac_stk_$1_$pac_stk_level])
-])
-ifelse([$2],[],[
-    dnl Save the last pushed value of VARNAME as pac_LastSavedValueOf_$VARNAME
-    PAS_VAR_COPY([pac_LastSavedValueOf_$1],[pac_stk_$1_$pac_stk_level])
-    dnl AS_ECHO(["pac_LastSavedValueOf_$1 = $pac_LastSavedValueOf_$1"])
-],[
-    dnl Save the last pushed value of VARNAME as $2
-    PAS_VAR_COPY([$2],[pac_stk_$1_$pac_stk_level])
-    dnl AS_ECHO(["$2 = $$2"])
-])
-AS_VAR_POPDEF([pac_stk_level])
-# END of PUSHVAL
-])
-dnl
-dnl
-dnl
-dnl PAC_VAR_POPVAL(VARNAME) 
-dnl
-dnl Restore variable, VARNAME, from the stack.
-dnl This macro is safe with respect to the nesting.
-dnl Some minimal checking of nesting balance of PAC_VAR_PUSH[POP]VAL()
-dnl is done here.
-dnl
-AC_DEFUN([PAC_VAR_POPVAL],[
-# START of POPVAL
-dnl define local m4-name pac_stk_level.
-AS_VAR_PUSHDEF([pac_stk_level], [pac_stk_$1_level])
-AS_VAR_SET_IF([pac_stk_level],[
-    AS_VAR_IF([pac_stk_level],[-1],[
-        AC_MSG_WARN(["Imbalance of PUSHVAL/POPVAL of $1"])
-    ],[
-        dnl AS_ECHO_N(["POPVAL: pac_stk_level = $pac_stk_level, "])
-        PAS_VAR_COPY([$1],[pac_stk_$1_$pac_stk_level])
-        dnl AS_ECHO(["popped_val = $$1"])
-        dnl autoconf < 2.64 does not have AS_VAR_ARITH, so use expr instead.
-        AS_VAR_SET([pac_stk_level], [`expr $pac_stk_level - 1`])
-    ])
-],[
-    AC_MSG_WARN(["Uninitialized PUSHVAL/POPVAL of $1"])
-])
-AS_VAR_POPDEF([pac_stk_level])
-# END of POPVAL
-])
-dnl
-dnl
-dnl
+dnl PAC_RUNLOG_IFELSE(COMMMAND,[ACTION-IF-RUN-OK],[ACTION-IF-RUN-FAIL])
+AC_DEFUN([PAC_RUNLOG_IFELSE],[AS_IF([AC_RUN_LOG([$1])],[$2],[$3])])
+
 dnl PAC_COMPILE_IFELSE_LOG is a wrapper around AC_COMPILE_IFELSE with the
 dnl output of ac_compile to a specified logfile instead of AS_MESSAGE_LOG_FD
 dnl
 dnl PAC_COMPILE_IFELSE_LOG(logfilename, input,
 dnl                        [action-if-true], [action-if-false])
 dnl
-dnl where input, [action-if-true] and [action-if-false] are used
-dnl in AC_COMPILE_IFELSE(input, [action-if-true], [action-if-false]).
-dnl This macro is nesting safe.
-dnl
 AC_DEFUN([PAC_COMPILE_IFELSE_LOG],[
-dnl
-dnl Instead of defining our own ac_compile and do AC_TRY_EVAL
-dnl on these variables.  We modify ac_compile used by AC_*_IFELSE
-dnl by piping the output of the command to a logfile.  The reason is that
-dnl 1) AC_TRY_EVAL is discouraged by Autoconf. 2) defining our ac_compile
-dnl could mess up the usage and order of *CFLAGS, LDFLAGS and LIBS in
-dnl these commands, i.e. deviate from how GNU standard uses these variables.
-dnl
-dnl Replace ">&AS_MESSAGE_LOG_FD" by "> FILE 2>&1" in ac_compile.
-dnl Save a copy of ac_compile on a stack
-dnl which is safe through nested invocations of this macro.
-PAC_VAR_PUSHVAL([ac_compile])
-dnl Modify ac_compile based on the unmodified ac_compile.
-ac_compile="`echo $pac_FirstSavedValueOf_ac_compile | sed -e 's|>.*$|> $1 2>\&1|g'`"
-AC_COMPILE_IFELSE([$2],[
-    ifelse([$3],[],[:],[$3])
-],[
-    ifelse([$4],[],[:],[$4])
-])
-dnl Restore the original ac_compile from the stack.
-PAC_VAR_POPVAL([ac_compile])
+PAC_PUSH_FLAG([ac_compile])
+ac_compile="`echo $ac_compile | sed -e 's|>.*$|> $1 2>\&1|g'`"
+AC_COMPILE_IFELSE([$2],[$3],[$4])
+PAC_POP_FLAG([ac_compile])
 ])
-dnl
-dnl
-dnl
+
 dnl PAC_LINK_IFELSE_LOG is a wrapper around AC_LINK_IFELSE with the
 dnl output of ac_link to a specified logfile instead of AS_MESSAGE_LOG_FD
 dnl
 dnl PAC_LINK_IFELSE_LOG(logfilename, input,
 dnl                     [action-if-true], [action-if-false])
 dnl
-dnl where input, [action-if-true] and [action-if-false] are used
-dnl in AC_LINK_IFELSE(input, [action-if-true], [action-if-false]).
-dnl This macro is nesting safe.
-dnl
 AC_DEFUN([PAC_LINK_IFELSE_LOG],[
-dnl
-dnl Instead of defining our own ac_link and do AC_TRY_EVAL
-dnl on these variables.  We modify ac_link used by AC_*_IFELSE
-dnl by piping the output of the command to a logfile.  The reason is that
-dnl 1) AC_TRY_EVAL is discouraged by Autoconf. 2) defining our ac_link
-dnl could mess up the usage and order of *CFLAGS, LDFLAGS and LIBS in
-dnl these commands, i.e. deviate from how GNU standard uses these variables.
-dnl
-dnl Replace ">&AS_MESSAGE_LOG_FD" by "> FILE 2>&1" in ac_link.
-dnl Save a copy of ac_link on a stack
-dnl which is safe through nested invocations of this macro.
-PAC_VAR_PUSHVAL([ac_link])
-dnl Modify ac_link based on the unmodified ac_link.
-ac_link="`echo $pac_FirstSavedValueOf_ac_link | sed -e 's|>.*$|> $1 2>\&1|g'`"
-dnl
-AC_LINK_IFELSE([$2],[
-    ifelse([$3],[],[:],[$3])
-],[
-    ifelse([$4],[],[:],[$4])
+PAC_PUSH_FLAG([ac_link])
+ac_link="`echo $ac_link | sed -e 's|>.*$|> $1 2>\&1|g'`"
+AC_LINK_IFELSE([$2],[$3],[$4])
+PAC_POP_FLAG([ac_link])
 ])
-dnl Restore the original ac_link from the stack.
-PAC_VAR_POPVAL([ac_link])
-])
-dnl
-dnl
-dnl
+
 dnl PAC_COMPLINK_IFELSE (input1, input2, [action-if-true], [action-if-false])
 dnl
 dnl where input1 and input2 are either AC_LANG_SOURCE or AC_LANG_PROGRAM
 dnl enclosed input programs.
 dnl
 dnl The macro first compiles input1 and uses the object file created
-dnl as part of LIBS during linking.  This macro is nesting safe.
+dnl as part of LIBS during linking.
 dnl
 AC_DEFUN([PAC_COMPLINK_IFELSE],[
 AC_COMPILE_IFELSE([$1],[
-    PAC_RUNLOG([mv conftest.$OBJEXT pac_conftest.$OBJEXT])
-    PAC_VAR_PUSHVAL([LIBS])
-    LIBS="pac_conftest.$OBJEXT $pac_FirstSavedValueOf_LIBS"
-    AC_LINK_IFELSE([$2],[
-        ifelse([$3],[],[:],[$3])
-    ],[
-        ifelse([$4],[],[:],[$4])
-    ])
-    PAC_VAR_POPVAL([LIBS])
-    rm -f pac_conftest.$OBJEXT
-],[
-    ifelse([$4],[],[:],[$4])
-])
+    AC_RUN_LOG([mv conftest.$OBJEXT pac_conftest.$OBJEXT])
+    PAC_PUSH_FLAG([LIBS])
+    LIBS="pac_conftest.$OBJEXT $LIBS"
+    AC_LINK_IFELSE([$2],[$3],[$4])
+    PAC_POP_FLAG([LIBS])
+    rm -f pac_conftest.$OBJEXT]),[$4]
 ])
-- 
1.8.1.2

